Terminal.Gui/Drawing/Color.cs

```diff
@@ -1,369 +1,155 @@
+		ArgumentOutOfRangeException.ThrowIfNegative ( blue, nameof ( blue ) );
+		ArgumentOutOfRangeException.ThrowIfNegative ( alpha, nameof ( alpha ) );
+
+		A = Convert.ToByte ( alpha );
+		R = Convert.ToByte ( red );
+		G = Convert.ToByte ( green );
+		B = Convert.ToByte ( blue );
+	}
+
+	/// <summary>
+	/// Initializes a new instance of the <see cref="Color"/> class with an encoded signed 32-bit color value in ARGB32 format.
+	/// </summary>
+	/// <param name="rgba">The encoded 32-bit color value (see <see cref="Rgba"/>).</param>
+	/// <remarks>The alpha channel is not currently supported, so the value of the alpha channel bits will not affect rendering.</remarks>
+	public Color ( int rgba )
 	{
-		R = red;
-		G = green;
-		B = blue;
-		A = alpha;
+		Rgba = rgba;
 	}
 
 	/// <summary>
-	/// Initializes a new instance of the <see cref="Color"/> class with an encoded 24-bit color value.
+	/// Initializes a new instance of the <see cref="Color"/> class with an encoded unsigned 32-bit color value in ARGB32 format.
 	/// </summary>
-	/// <param name="rgba">The encoded 24-bit color value (see <see cref="Rgba"/>).</param>
-	public Color (int rgba)
+	/// <param name="argb">The encoded unsigned 32-bit color value (see <see cref="Argb"/>).</param>
+	/// <remarks>The alpha channel is not currently supported, so the value of the alpha channel bits will not affect rendering.</remarks>
+	public Color ( uint argb )
 	{
-		A = (byte)(rgba >> 24 & 0xFF);
-		R = (byte)(rgba >> 16 & 0xFF);
-		G = (byte)(rgba >> 8 & 0xFF);
-		B = (byte)(rgba & 0xFF);
+		Argb = argb;
 	}
 
 	/// <summary>
-	/// Initializes a new instance of the <see cref="Color"/> color from a legacy 16-color value.
+	/// Initializes a new instance of the <see cref="Color"/> color from a legacy 16-color named value.
 	/// </summary>
 	/// <param name="colorName">The 16-color value.</param>
-	public Color (ColorName colorName)
+	public Color ( in ColorName colorName )
 	{
-		var c = FromColorName (colorName);
-		R = c.R;
-		G = c.G;
-		B = c.B;
-		A = c.A;
+		this = ColorExtensions.ColorNameToColorMap [ colorName ];
 	}
 
 	/// <summary>
-	/// Initializes a new instance of the <see cref="Color"/> color from string. See <see cref="TryParse(string, out Color)"/>
+	/// Initializes a new instance of the <see cref="Color"/> color from string. See <see cref="TryParse(string, out Color?)"/>
 	/// for details.
 	/// </summary>
 	/// <param name="colorString"></param>
-	/// <exception cref="Exception"></exception>
-	public Color (string colorString)
+	/// <exception cref="ArgumentNullException">If <paramref name="colorString"/> is <see langword="null"/>.</exception>
+	/// <exception cref="ArgumentException">If <paramref name="colorString"/> is an empty string or consists of only whitespace characters.</exception>
+	/// <exception cref="ColorParseException">If thrown by <see cref="Parse(string?,System.IFormatProvider?)"/></exception>
+	public Color ( string colorString )
```

Comment:
Please hit CTRL-K CTRL-D in all files you touch. Spaces inside parens are counter to the rest of our formatting standards.

Reply:
I'm making edits to the resharper dotsettings to bring things like that in line with the editorconfig, since it's clearly failing to respect it the way it should. ReSharper owns the formatting in my visual studio setup, so making that match will take care of this. :)

Reply:
I now have a dotsettings (included in the upcoming change set) which looks to me like it resulted in the desired formatting. Let me know if you notice anything else.